### PR TITLE
Boost: Update deferjs to exclude application/json

### LINKED ISSUE #19158: Podcast Player Block: Player not appearing
#### Steps to reproduce the issue

1. Go to the editor for a page or post
2. Add a Podcast Player block
3. Check the preview or published page

#### What I expected

I would expect to see the embedded player on the published page as it appears in the editor.

#### What happened instead

I only see the podcast information, but no player. The player opens with a link in a different tab. 

**Screenshots**

Editor: https://d.pr/i/ojZHrz
Published page: https://d.pr/i/onL6ex

### PR DESCRIPTION
**Pull request by karthikax**
Fixes #19158

#### Changes proposed in this Pull Request:

* Adds `application/json` script type to the exclusion logic while deferring Javascript.

#### Jetpack product discussion

NA

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Activate Jetpack
* Activate Boost and its "Defer Non-Essential JavaScript" option
* Go to Posts > Add New and add a new Podcast block
* Paste in a podcast URL such as https://distributed.blog/category/podcast/feed/
* Publish your post
* Ensure that the player works well in the editor as well as in the front end.

### CODE REVIEW

File: projects/plugins/boost/app/features/optimizations/render-blocking-js/class-render-blocking-js.php [hunk 1]
```diff
@@ -198,17 +198,25 @@ protected function get_script_tags( $buffer ) {
...
-		return preg_replace_callback(
+	protected function ignore_exclusion_scripts( $buffer ) {
+		$exclusions = array(
+			// Scripts inside HTML comments.
 			'~<!--.*?-->~si',
-			function ( $comment_match ) {
-				return str_replace( '<script', sprintf( '<script %s="%s"', esc_html( $this->ignore_attribute ), esc_attr( $this->ignore_value ) ), $comment_match[0] );
+
+			// Scripts with application/json type
+			'~<script.*type="application/json".*>.*</script>~si',
```
Comment (thingalon) [inline projects/plugins/boost/app/features/optimizations/render-blocking-js/class-render-blocking-js.php:213]: The HTML spec allows single-quotes or no quotes here.

You can detect these cases using a regexp like this:
```suggestion
			'~<script.*type=(?<q>["\']*)application/json\k<q>*>.*</script>~si',
```

**Review by thingalon — CHANGES_REQUESTED** [review]
Looks good, except I think the regexp could be more flexible with quotes.

File: projects/plugins/boost/app/features/optimizations/render-blocking-js/class-render-blocking-js.php (hunk 1 above)
Comment (karthikax) [reply projects/plugins/boost/app/features/optimizations/render-blocking-js/class-render-blocking-js.php]: Thanks for the suggestion. Updated this now.

**Review by thingalon — APPROVED** [review]
👍 Thanks karthik

**Pull request merged**